y/metrics.go

```diff
@@ -25,67 +25,121 @@ var (
-	numWrites *expvar.Int
-	// numBytesRead has cumulative number of bytes read
-	numBytesRead *expvar.Int
-	// numBytesWritten has cumulative number of bytes written
-	numBytesWritten *expvar.Int
-	// numLSMGets is number of LMS gets
+	// VLOG METRICS
+	// numReads has cumulative number of reads from vlog
+	numReadsVlog *expvar.Int
+	// numWrites has cumulative number of writes into vlog
+	numWritesVlog *expvar.Int
+	// numBytesRead has cumulative number of bytes read from VLOG
+	numBytesReadVlog *expvar.Int
+	// numBytesVlogWritten has cumulative number of bytes written into VLOG
+	numBytesVlogWritten *expvar.Int
+
+	// LSM METRICS
+	// numBytesRead has cumulative number of bytes read from LSM tree
+	numBytesReadLSM *expvar.Int
+	// numBytesLSMWritten has cumulative number of bytes written into LSM Tree
+	numBytesLSMWritten *expvar.Int
+	// numLSMGets is number of LSM gets
 	numLSMGets *expvar.Map
+	// numBytesCompactionWritten is the number of bytes written in the lsm tree due to compaction
+	numBytesCompactionWritten *expvar.Int
 	// numLSMBloomHits is number of LMS bloom hits
 	numLSMBloomHits *expvar.Map
-	// numGets is number of gets
+
+	// DB METRICS
+	// numGets is number of gets -> Number of get requests made
 	numGets *expvar.Int
-	// numPuts is number of puts
+	// number of get queries in which we actually get a result
+	numGetsWithResults *expvar.Int
+	// number of iterators created, these would be the number of range queries
+	numIteratorsCreated *expvar.Int
+	// numPuts is number of puts -> Number of puts requests made
 	numPuts *expvar.Int
-	// numBlockedPuts is number of blocked puts
-	numBlockedPuts *expvar.Int
-	// numMemtableGets is number of memtable gets
+	// numMemtableGets is number of memtable gets -> Number of get requests made on memtable
 	numMemtableGets *expvar.Int
 	// numCompactionTables is the number of tables being compacted
 	numCompactionTables *expvar.Int
+	// Total writes by a user in bytes
+	numBytesWrittenUser *expvar.Int
 )
 
 // These variables are global and have cumulative values for all kv stores.
 func init() {
-	numReads = expvar.NewInt("badger_v3_disk_reads_total")
-	numWrites = expvar.NewInt("badger_v3_disk_writes_total")
-	numBytesRead = expvar.NewInt("badger_v3_read_bytes")
-	numBytesWritten = expvar.NewInt("badger_v3_written_bytes")
-	numLSMGets = expvar.NewMap("badger_v3_lsm_level_gets_total")
-	numLSMBloomHits = expvar.NewMap("badger_v3_lsm_bloom_hits_total")
-	numGets = expvar.NewInt("badger_v3_gets_total")
-	numPuts = expvar.NewInt("badger_v3_puts_total")
-	numBlockedPuts = expvar.NewInt("badger_v3_blocked_puts_total")
-	numMemtableGets = expvar.NewInt("badger_v3_memtable_gets_total")
-	lsmSize = expvar.NewMap("badger_v3_lsm_size_bytes")
-	vlogSize = expvar.NewMap("badger_v3_vlog_size_bytes")
-	pendingWrites = expvar.NewMap("badger_v3_pending_writes_total")
-	numCompactionTables = expvar.NewInt("badger_v3_compactions_current")
+	numReadsVlog = expvar.NewInt("badger_v4_disk_reads_total")
```

Comment:
nit/maintenance: can we move the prefix `badger_v4` in a constant `BADGER_METRIC_PREFIX` and concat here, its a one time expense to do the string concat. It makes it